### PR TITLE
Rubic 706/add optional receiver address

### PR DESCRIPTION
**Pull request by Doctor-Cyclone**

### CODE REVIEW

File: src/core/blockchain/web3-public/web3-public.ts [hunk 1]
```diff
@@ -286,6 +286,9 @@ export class Web3Public {
         address: string,
         tokensAddresses: string[]
     ): Promise<BigNumber[]> {
+        if (MULTICALL_ADDRESSES[this.blockchainName]) {
```
Comment (axtezy) [inline src/core/blockchain/web3-public/web3-public.ts:289]: Это всегда тру будет, нужно сравнить с 'NONE'

File: src/features/cross-chain/providers/rango-trade-provider/rango-cross-chain-trade-provider.ts [hunk 2]
```diff
@@ -79,11 +80,17 @@ export class RangoCrossChainTradeProvider extends CrossChainTradeProvider {
         return false;
     }
 
+    // TODO Reduce complexity
```
Comment (axtezy) [inline src/features/cross-chain/providers/rango-trade-provider/rango-cross-chain-trade-provider.ts:83]: @TODO

File: src/features/instant-trades/dexes/common/oneinch-common/oneinch-trade.ts [hunk 3]
```diff
@@ -140,7 +140,11 @@ export class OneinchTrade extends InstantTrade {
         await this.checkAllowanceAndApprove(options);
 
         try {
-            const apiTradeData = await this.getTradeData().catch(err => {
+            const apiTradeData = await this.getTradeData(
+                undefined,
```
Comment (axtezy) [inline src/features/instant-trades/dexes/common/oneinch-common/oneinch-trade.ts:144]: false

**Review by axtezy — CHANGES_REQUESTED** [review]

File: src/core/blockchain/web3-public/web3-public.ts (hunk 1 above)
Comment (Doctor-Cyclone) [reply src/core/blockchain/web3-public/web3-public.ts]: Done.

File: src/features/cross-chain/providers/rango-trade-provider/rango-cross-chain-trade-provider.ts (hunk 2 above)
Comment (Doctor-Cyclone) [reply src/features/cross-chain/providers/rango-trade-provider/rango-cross-chain-trade-provider.ts]: Done.

File: src/features/instant-trades/dexes/common/oneinch-common/oneinch-trade.ts (hunk 3 above)
Comment (Doctor-Cyclone) [reply src/features/instant-trades/dexes/common/oneinch-common/oneinch-trade.ts]: Done.

**Review by kolebayevv — APPROVED** [review]

File: src/features/cross-chain/providers/rango-trade-provider/rango-cross-chain-trade.ts [hunk 4]
```diff
@@ -160,6 +164,10 @@ export class RangoCrossChainTrade extends CrossChainTrade {
     }
 
     public async swap(options: SwapTransactionOptions = {}): Promise<string> {
+        if (options?.receiverAddress) {
+            throw new RubicSdkError("Selected provider doesn't support receiver address");
```
Comment (ottebrut) [inline src/features/cross-chain/providers/rango-trade-provider/rango-cross-chain-trade.ts:168]: throw new UnsupportedReceiverAddressError();

File: src/features/instant-trades/dexes/common/lifi/lifi-trade.ts [hunk 5]
```diff
@@ -111,6 +111,10 @@ export class LifiTrade extends InstantTrade {
     }
 
     public async swap(options: SwapTransactionOptions = {}): Promise<TransactionReceipt> {
+        if (options?.receiverAddress) {
+            throw new RubicSdkError("Selected provider doesn't support receiver address");
```
Comment (ottebrut) [inline src/features/instant-trades/dexes/common/lifi/lifi-trade.ts:115]: throw new UnsupportedReceiverAddressError();

File: src/features/instant-trades/dexes/common/zrx-common/zrx-trade.ts [hunk 6]
```diff
@@ -54,6 +55,10 @@ export class ZrxTrade extends InstantTrade {
     }
 
     public async swap(options: SwapTransactionOptions = {}): Promise<TransactionReceipt> {
+        if (options?.receiverAddress) {
+            throw new RubicSdkError("Selected provider doesn't support receiver address");
```
Comment (ottebrut) [inline src/features/instant-trades/dexes/common/zrx-common/zrx-trade.ts:59]: throw new UnsupportedReceiverAddressError();

**Review by ottebrut — CHANGES_REQUESTED** [review]

**Review by axtezy — APPROVED** [review]

File: src/features/cross-chain/providers/rango-trade-provider/rango-cross-chain-trade.ts (hunk 4 above)
Comment (Doctor-Cyclone) [reply src/features/cross-chain/providers/rango-trade-provider/rango-cross-chain-trade.ts]: Done.

File: src/features/instant-trades/dexes/common/lifi/lifi-trade.ts (hunk 5 above)
Comment (Doctor-Cyclone) [reply src/features/instant-trades/dexes/common/lifi/lifi-trade.ts]: Done.

File: src/features/instant-trades/dexes/common/zrx-common/zrx-trade.ts (hunk 6 above)
Comment (Doctor-Cyclone) [reply src/features/instant-trades/dexes/common/zrx-common/zrx-trade.ts]: Done.

**Review by ottebrut — APPROVED** [review]

**Pull request merged**